Camera2SlowMotion/app/src/main/java/com/example/android/camera2/slowmo/AnalyserActivity.java

```diff
@@ -0,0 +1,125 @@
+  private LagCalculator lagCalculator;
+  private ResultPublisher resultPublisher;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.activity_analyser);
+
+    filePath = findViewById(R.id.selectedFilePath);
+    filePickerBtn = findViewById(R.id.fileSelectButton);
+    analyzeBtn = findViewById(R.id.analyze);
+    analyseResultField = findViewById(R.id.analyseResultView);
+
+    analyseResultField.setMovementMethod(new ScrollingMovementMethod());
+
+    videoProcessor = new VideoProcessor();
+
+    lagCalculator = new LagCalculator(videoProcessor, CameraActivity.serverHandler);
+
+    resultPublisher = new ResultPublisher();
+
+    analyseResultField.append(
+        "\nPort bound:" + CameraActivity.serverHandler.getServerSocketPort() + "\n");
+
+    CameraFragment.Companion.getFilePath();
+    filePath.setText(CameraFragment.Companion.getFilePath());
+
+    filePickerBtn.setOnClickListener(
+        new View.OnClickListener() {
+          @Override
+          public void onClick(View view) {
+            fileIntent = new Intent(Intent.ACTION_GET_CONTENT);
+            fileIntent.setType("*/*");
+            startActivityForResult(fileIntent, FILE_PICKER_REQUEST_CODE);
+          }
+        });
+
+    analyzeBtn.setOnClickListener(
+        new View.OnClickListener() {
+          @RequiresApi(api = Build.VERSION_CODES.P)
+          @Override
+          public void onClick(View view) {
+            try {
+              analyze();
+            } catch (Exception e) {
+              Log.d(ContentValues.TAG, "Analyse call failed with: " + e.getMessage());
+            }
+          }
+        });
+  }
+
+  @Override
+  protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+    super.onActivityResult(requestCode, resultCode, data);
+    if (requestCode == FILE_PICKER_REQUEST_CODE && resultCode == RESULT_OK) {
+      String path = data.getData().toString();
+      fileUri = data.getData();
+      filePath.setText(path);
+    }
+  }
+
+  /** Triggers subsequent lifecycle events of lag calculations. */
+  @RequiresApi(api = Build.VERSION_CODES.P)
+  private void analyze() {
+    fileUri = Uri.parse(filePath.getText().toString());
+    videoProcessor.setVideoReader(getApplicationContext(), fileUri);
+
+    CameraActivity.serverHandler.downloadServerTimeStamps();
+
+    videoProcessor.doOcrOnVideo(lagCalculator);
+
+    resultPublisher.execute(lagCalculator);
+  }
+
+  /** A background threads which listens for the results of LagCalculator and shows the results
+   * when available to TextView.*/
+  private class ResultPublisher extends AsyncTask<LagCalculator, Void, Void> {
+    private Integer resultPublishedCount = 0;
+
+    @Override
+    protected Void doInBackground(LagCalculator... lagCalculators) {
+      while (true) {
+        if (resultPublishedCount < lagCalculators[0].lagResult.size()) {
```

Comment:
You are only using the 0th element:
1. How do you know the list isn't empty?
2. Why aren't you using the rest?

If it is only a single object, please try to pass a single object rather than an array of size 1, and if not possible, please document why it is not possible, and the answers to the above two points.